list/directory/list.js

```diff
@@ -0,0 +1,439 @@
+                        return regExp.test(obj[this.props.fieldName]);
+                    });
+                })();
+            }
+        }else{
+            dataToUse = this.sortedData;
+        }
+        return (() => {
+            return groupBy(dataToUse, (line) => {
+                return line[this.props.fieldName][0].toUpperCase();
+            });
+        })();
+    },
+
+    /**
+     * Get one group by letter.
+     * @param {string} selected letter
+     * @returns {string} group corresponding to selected letter
+     * @private
+     */
+    _getGroupFromData(dataToUse, letter){
+        var group = {};
+        group[letter] = dataToUse[letter];
+        return group;
+    },
+
+    /**
+     * Get initial letter from data.
+     * Hint: The grouped array is already sorted, so we need to pick only first object property name.
+     * @param {object} grouped data. exemple: {A: Array[], B: Array[], C: Array[], ..., Z: Array[]}
+     * @returns {string} initial selected data from grouped array
+     * @private
+     */
+    _getInitialSelectedLetter(data){
+        return keys(data)[0];
+    },
+
+    /**
+     * Retrieve available letters from data.
+     * Hint: The grouped data is already sorted by letter.
+     * @param {object} grouped data. exemple: {A: Array[], B: Array[], C: Array[], ..., Z: Array[]}
+     * @returns {string} a string of available letters
+     * @private
+     */
+    _getAvailableLetters(data){
+        return keys(data).join('');
+    },
+
+    /**
+     * Handle letter click event.
+     * if showAll is true, scroll to letter group. Otherwise, show only group related to selected letter
+     * @param {string} selected letter
+     * @private
+     */
+    _onLetterSelected(letter){
+        if(this.props.showAll){//display all data
+            var groupView = document.getElementById(this._getGroupId(letter));
+            groupView.scrollIntoView();
+        }else{//display only one group
+            var dataToUse = this._getDataToUse();
+            var availableLetters = this._getAvailableLetters(dataToUse);
+            var groupToDisplay = this._getGroupFromData(dataToUse, letter);
+            this.setState({data: groupToDisplay, currentSelectedLetter: letter, availableLetters: availableLetters});
+        }
+
+    },
+
+    /**
+     * Filter results by searching text in data.
+     * @private
+     */
+    _filterResults(){
+        this.searchText = this.quickSearchDOMNode.value.trim();
+        var newState = this._getState();
+
+        this.setState(
+            {   data: newState.data,
+                currentSelectedLetter: newState.currentSelectedLetter,
+                availableLetters: newState.availableLetters
+            });
+    },
+
+    /**
+     * Get group id.
+     * @param {string} selected letter
+     * @returns {string} group id
+     * @private
+     */
+    _getGroupId(letter){
+        return `directory-group-${letter}`;
```

Comment:
use a data attribute instead of an id `data-directory-group='${letter}'`, this way the selector is always the same, only the value changes.